core/src/main/java/nu/yona/server/device/entities/DeviceAnonymized.java

```diff
@@ -156,6 +156,7 @@ public void addVpnStatusChangeEvent(VpnStatusChangeEvent event)
 	{
 		vpnStatusChangeEvents.add(event);
 		event.setDeviceAnonymized(this);
+		lastVpnStatusChangeEvent = event;
```

Comment:
This fix or change was not mentioned in the commit message

Reply:
This was introduced with commit d242b76836a8b48e723e102939685294a8892e0f. That commit indeed does not mention this change. The change is made because we now read the last VPN status change event from the ``DeviceAnonymized`` entity that was saved to the database. The field ``lastVpnStatusChangeEvent`` is defined this way:
```
	@ManyToOne
	@JoinColumnsOrFormulas({
			@JoinColumnOrFormula(formula = @JoinFormula(value = "(SELECT e.id FROM vpn_status_change_events e WHERE e.device_anonymized_id = id ORDER BY e.id DESC LIMIT 1)", referencedColumnName = "id")) })
	private VpnStatusChangeEvent lastVpnStatusChangeEvent;
```
This implies that the field is initialized when loading the entity from the database, but to be able to use it before or after save, the code needs to explicitly initialize it, similar to what we do in two-way associations. See e.g. ``UserAnonymized.addDeviceAnonymized``:
```
		devicesAnonymized.add(deviceAnonimized);
		deviceAnonimized.setUserAnonymized(this);
```
The first line is added for the same reason as the new line in ``DeviceAnonymized.addVpnStatusChangeEvent``.